library: Add Scale entry

SoNiC-HeRE opened the pull request:
Added Scale Entry under User Interface Section in Library
Issue #69

sonnyp reviewed (changes requested):
I think we should add a vertical example and have one of them with `draw-value: true`

Let's use the `UI` panel whenever possible. I know it's not obvious but you can do most of this directly from there , hopefully Workbench will make that more obvious to new developers.

example:

```
    Scale one {
      orientation: horizontal;
      margin-bottom: 20;
      adjustment: Gtk.Adjustment {
        lower: 0;
        upper: 100;
        value: 70;
      };
    }
```

SoNiC-HeRE commented on src/Library/demos/Scale/main.blp:
```diff
@@ -0,0 +1,44 @@
+
+Adw.StatusPage {
+  title: "Scale";
+  description: _("A simple sliding control");
```
👍

SoNiC-HeRE commented on src/Library/demos/Scale/main.js:
```diff
@@ -0,0 +1,36 @@
+scale_two.add_mark(20.0, "right", null);
+scale_two.add_mark(30.0, "right", null);
+scale_two.add_mark(40.0, "right", null);
+scale_two.add_mark(50.0, "right", null);
```
Okay, I'll add 2 marks with labels

SoNiC-HeRE commented on src/Library/demos/Scale/main.js:
```diff
@@ -0,0 +1,36 @@
+scale_one.set_range(true, 50);
+
+scale_two.set_range(true, 50);
+scale_two.set_inverted(true);
```
okay

SoNiC-HeRE commented on src/Library/demos/Scale/main.js:
```diff
@@ -0,0 +1,36 @@
+  }
+});
+
+scale_two.connect("value_changed", () => {
+  let scale_value_two = scale_two.get_value();
+  if (scale_value_two === 50) {
+    console.log("Max Value Reached");
+  } else if (scale_value_two === 1) {
+    console.log("Min Value Reached");
+  }
+});
```
Ok

SoNiC-HeRE commented:
Applied the Changes as Suggested , Implemented a Vertical Slider and stuck to a 3 slider layout for cleaner UI. Implemented 4 marks on the Vertical Slider and Updated JS Code

sonnyp reviewed (changes requested):
not building/working

please review your own changes before submitting for review

sonnyp commented on src/Library/demos/Scale/main.js:
```diff
@@ -0,0 +1,33 @@
+scale_two.add_mark(100.0, "right", "D");
+scale_two.set_increments(25.0, 25.0);
+
+scale_disabled.set_show_fill_level(25);
```
`set_show_fill_level` takes a boolean, not an integer
makes sure that you're using the APIs correctly—not just that things work

also `show-fille-level` is a property so it can be moved to `main.blp` if needed

SoNiC-HeRE replied:
okay, I'll move it to main.blp

sonnyp commented on src/Library/demos/Scale/main.js:
```diff
@@ -0,0 +1,33 @@
+scale_one.connect("value-changed", () => {
+  let scale_value = scale_one.get_value();
+  if (scale_value === scale_one.adjustment.upper) {
+    console.log("Max Value Reached");
```
```suggestion
    console.log("Maximum value reached");
```

sonnyp commented on src/Library/demos/Scale/main.js:
```diff
@@ -0,0 +1,33 @@
+  if (scale_value === scale_one.adjustment.upper) {
+    console.log("Max Value Reached");
+  } else if (scale_value === scale_one.adjustment.lower) {
+    console.log("Min Value Reached");
```
```suggestion
    console.log("Minimum value reached");
```

sonnyp commented on src/Library/demos/Scale/main.js:
```diff
@@ -0,0 +1,33 @@
+const scale_two = workbench.builder.get_object("two");
+const scale_disabled = workbench.builder.get_object("disabled");
+
+scale_two.add_mark(25.0, "left", "A");
+scale_two.add_mark(50.0, "right", "B");
+scale_two.add_mark(75.0, "right", "C");
+scale_two.add_mark(100.0, "right", "D");
```
ha yes - Blueprint doesn't support marks unfortunaly so we have to do it in JS

https://gitlab.gnome.org/jwestman/blueprint-compiler/-/issues/62

SoNiC-HeRE replied:
It's already in JS. So shall i leave it unchanged?

sonnyp replied:
yes - it's good as is

sonnyp commented on src/Library/demos/Scale/main.js:
```diff
@@ -0,0 +1,33 @@
+  }
+});
+
+scale_two.connect("value-changed", () => {
+  let scale_value_two = scale_two.get_value();
+  if (scale_value_two === 25) {
+    console.log("Mark A Reached");
+  } else if (scale_value_two === 50) {
+    console.log("Mark B Reached");
+  } else if (scale_value_two === 75) {
+    console.log("Mark C Reached");
+  } else if (scale_value_two === 100) {
+    console.log("Mark D Reached");
+  }
+});
```
Instead of repeating the mark values - please create them and find them from the same values.

For example - store the marks in an object and use that to add the marks and find the label in `value-chaged`

SoNiC-HeRE commented:
Applied Suggested Changes: Updated JS Code and implemented Changes in Blp File as well.

sonnyp reviewed (approved):
Thanks.

Regarding marks, this is what I meant: https://github.com/sonnyp/Workbench/pull/204/commits/4f8e9c54036b8c9904a6555d86a18019b1978bec
